src/Rules/Violations.php

```diff
@@ -108,13 +108,13 @@ public function toArray(): array
 
     public function remove(self $violations): void
     {
-        $this->violations = array_udiff(
+        $this->violations = array_values(array_udiff(
             $this->violations,
             $violations->violations,
-            function (Violation $a, Violation $b): int {
-                return $a == $b ? 0 : 1;
+            static function (Violation $a, Violation $b): int {
+                return $a->isEqualsTo($b) ? 0 : -1;
```

Comment:
```suggestion
                return $a <=> $b;
```

this should do the trick without a custom comparison function. would you mind trying it?

Reply:
I changed that to use the spaceship operator